nemo/collections/tts/data/dataset.py

```diff
@@ -483,6 +490,15 @@ def add_energy(self, **kwargs):
     def add_speaker_id(self, **kwargs):
         pass
 
+    def add_gst_ref_audio(self, **kwargs):
+        if SpeakerID not in self.sup_data_types_set:
+            self.speaker_to_index = None
+        else:
+            self.speaker_to_index = defaultdict(list)
+            for i, d in enumerate(self.data):
+                self.speaker_to_index[d.get('speaker_id', None)].append(i)
+            self.speaker_to_index = {k: set(v) for k, v in self.speaker_to_index.items()}
```

Comment:
Make the defaultdict values as set and not list so we do not have to make another pass to convert list to set.